Source/WebCore/dom/SimulatedClick.cpp

```diff
@@ -77,11 +79,55 @@ class SimulatedMouseEvent final : public MouseEvent {
     }
 };
 
+class SimulatedPointerEvent final : public PointerEvent {
+    WTF_MAKE_TZONE_OR_ISO_ALLOCATED_INLINE(SimulatedPointerEvent);
+public:
+    static RefPtr<SimulatedPointerEvent> create(const AtomString& type, const SimulatedMouseEvent& event, RefPtr<Event>&& underlyingEvent, Element& target, SimulatedClickSource source)
+    {
+        return adoptRef(*new SimulatedPointerEvent(type, event, WTFMove(underlyingEvent), target, source));
+    }
+
+private:
+    // If the device type cannot be detected by the user agent, then the value MUST be an empty string.
```

Comment:
Are these comments from the spec? If so, might be nice to add a spec link above line 82.

Reply:
they are; will do